src/t_string.c

```diff
@@ -384,30 +384,24 @@ void psetexCommand(client *c) {
     setGenericCommand(c, OBJ_PX | OBJ_ARGV3, c->argv[1], c->argv[3], c->argv[2], UNIT_MILLISECONDS, NULL, NULL, NULL);
 }
 
+/* DELIFEQ key value */
 void delifeqCommand(client *c) {
-    robj *existing_value = lookupKeyWrite(c->db, c->argv[1]);
-
-    if (existing_value == NULL) {
-        addReplyLongLong(c, 0);
-        return;
-    }
-
-    if (checkType(c, existing_value, OBJ_STRING)) {
-        /* Error reply already sent */
-        return;
-    }
+    robj *o;
+    if ((o = lookupKeyWriteOrReply(c, c->argv[1], shared.czero)) == NULL || checkType(c, o, OBJ_STRING)) return;
 
-    if (compareStringObjects(existing_value, c->argv[2]) != 0) {
-        addReplyLongLong(c, 0);
+    if (compareStringObjects(o, c->argv[2]) != 0) {
+        addReply(c, shared.czero);
         return;
     }
 
-    if (!dbSyncDelete(c->db, c->argv[1])) {
-        addReplyLongLong(c, 0);
-        return;
-    }
+    serverAssert(dbSyncDelete(c->db, c->argv[1]));
 
-    addReplyLongLong(c, 1);
+    /* Propagate as DEL command */
+    rewriteClientCommandVector(c, 2, shared.del, c->argv[1]);
+    signalModifiedKey(c, c->db, c->argv[1]);
+    notifyKeyspaceEvent(NOTIFY_GENERIC, "del", c->argv[1], c->db->id);
+    server.dirty++;
+    addReply(c, shared.cone);
```

Comment:
Five very good points. Thanks!

Just some explanations:

1. Replicate as DEL is important for compatibility with replicas running an older Valkey version.
2. Signal modified key is for WATCH and do invalidate client-side caching (client tracking)
3. Keyspace notifications
4. Dirty++ indicates there are unsaved changes
5. Using shared strings for the RESP zero and one replies avoids some allocation and string processing.

Reply:
about the No5, we actually have the fast paths:
```
void addReplyLongLong(client *c, long long ll) {
    if (ll == 0)
        addReply(c, shared.czero);
    else if (ll == 1)
        addReply(c, shared.cone);
    else {
        if (prepareClientToWrite(c) != C_OK) return;
        _addReplyLongLongWithPrefix(c, ll, ':');
    }
}
```